### PR TITLE
[pkg/telemetryquerylanguage] Add split function to TQL

### PR DESCRIPTION
**Pull request by linjunzhe**
**Description:** <Describe what has changed.>
Add `split` function to convert a string attribute into an array of substrings split by the delimiter. The scenario is also described in issue https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/11790

**Link to tracking Issue:**  https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/11790

**Testing:** 
Updated unit tests

**Documentation:** 
Updated readme in tqlotel

### CODE REVIEW

File: pkg/telemetryquerylanguage/functions/tqlotel/func_split.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/telemetryquerylanguage/tql"
+)
+
+func Split(target tql.GetSetter, delimiter string) (tql.ExprFunc, error) {
+	return func(ctx tql.TransformContext) interface{} {
+		if val := target.Get(ctx); val != nil {
+			if valStr, ok := val.(string); ok {
+				strSlice := strings.Split(valStr, delimiter)
+				if len(strSlice) > 0 {
+					target.Set(ctx, strSlice)
```
[the remark this answers is not in the document]
Comment (TylerHelmuth) [reply pkg/telemetryquerylanguage/functions/tqlotel/func_split.go:29]: SpanID and TraceID are examples of factory functions.

[the remark this answers is not in the document]
Comment (linjunzhe) [reply pkg/telemetryquerylanguage/functions/tqlotel/func_split.go:29]: Good point. Sorry for the late response. I can change it to factory function.

File: pkg/telemetryquerylanguage/functions/tqlotel/func_split_test.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+import (
+	"testing"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/telemetryquerylanguage/tql"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/telemetryquerylanguage/tql/tqltest"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_split(t *testing.T) {
```
Comment (TylerHelmuth) [inline pkg/telemetryquerylanguage/functions/tqlotel/func_split_test.go:25]: Can you add a test case for empty string and empty delimiter?

Comment (linjunzhe) [reply pkg/telemetryquerylanguage/functions/tqlotel/func_split_test.go:25]: @TylerHelmuth Yes. 
* https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e4acc8d1b4b21968748fff97d382edafb9a9548e/pkg/telemetryquerylanguage/functions/tqlotel/func_split_test.go#L43-L51 For empty string 
* https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e4acc8d1b4b21968748fff97d382edafb9a9548e/pkg/telemetryquerylanguage/functions/tqlotel/func_split_test.go#L53-L61 For empty delimiter

Comment (TylerHelmuth) [reply pkg/telemetryquerylanguage/functions/tqlotel/func_split_test.go:25]: @linjunzhe sorry I meant a single test where both the string and the delimiter are empty string at the same time.

Comment (linjunzhe) [reply pkg/telemetryquerylanguage/functions/tqlotel/func_split_test.go:25]: @TylerHelmuth Sure. Added the test case.

**Comment by linjunzhe** [comment]
@TylerHelmuth Could you help to review the code again? What else I need to do to get the approval and code merged.

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**